torch/csrc/dynamo/eval_frame.c

```diff
@@ -322,12 +348,126 @@ static void destroy_cache_entry(CacheEntry* e) {
   free(e);
 }
 
-inline static CacheEntry* get_cache_entry(PyCodeObject* code) {
-  CacheEntry* extra = NULL;
-  _PyCode_GetExtra((PyObject*)code, cache_entry_extra_index, (void*)&extra);
+/* CacheEntry helper functions ends */
+
+/* Extractions helper functions begins. They help with NULL and SKIP_CODE corner cases */
+
+inline static CacheEntry* extract_cache_entry(ExtraState* extra_state) {
+  // Helper to extra the cache_entry from the extra state.
+
+  // Ownership contract
+  // args
+  //  - extra_state: Borrowed
+  // return
+  //  - CacheEntry: Borrowed.
+  if (extra_state == NULL || extra_state == SKIP_CODE) {
+    return NULL;
+  }
+  return extra_state->cache_entry;
+}
+
+
+inline static FrameState* extract_frame_state(ExtraState* extra_state) {
+  // Returns either the previously stored frame state or an empty dict.
+
+  // Ownership contract
+  // args
+  //  - extra_state: Borrowed
+  // return
+  //  - extra_state->frame_state: Borrowed.
+  if (extra_state == NULL || extra_state == SKIP_CODE) {
+    return NULL;
+  }
+  return extra_state->frame_state;
+}
+
+/* Extractions helper functions ends */
+
+/* Extra state helper functions begins */
+
+inline static ExtraState* get_extra_state(PyCodeObject* code) {
+  // Ownership contract
+  // args
+  //  - code: Borrowed
+  // return
+  //  - extra_state: Borrowed. set_extra_state is the owner.
```

Comment:
I still don't know what it means for a function to "own" data

Reply:
I will just remove that line. What I really meant here was the set_extra_State is responsible for cleaning it up, but it does not make sense here.